src/util/remove-undefined-properties.ts

```diff
@@ -0,0 +1,9 @@
+export function removeUndefinedProperties(obj: any): any {
+  const copy = Object.assign({}, obj);
+  for (const key in copy) {
+    if (copy[key] === undefined) {
+      delete copy[key];
+    }
+  }
+  return copy;
+}
```

Comment:
I'm a bit concerned about the performance implications here, the `endpoing.merge()` method gets called a lot. The `headers` object is already a copy via lowercaseKeys, so no need to create a new copy for that again.

Maybe remove the copy line from the method, and instead pass a copy into it when needed? Like this

```diff
- options = removeUndefinedProperties(options);
+ options = removeUndefinedProperties(Object.assign({}, options));
```

I'd also be curious to learn about the performance of creating a shallow copy and then removing undefined keys, vs creating a new object and only adding the properties to it that are not undefined. Could you maybe setup such a test? https://jsbench.me/ is good for this kind of test. I'd define two methods for the two startegies, and then test each twice: once with an object without any undefined properties, and once with some undefined properties

Reply:
The fastest option was to delete undefined properties from an existing object, so I'll switch to that approach. https://jsbench.me/78kfvy50sd/2

Reply:
Note that it's not exactly the same thing as in the benchmark because the lowercasing function is already doing a copy with undefined properties included. I didn't want to change that.

Reply:
Thanks Francisco, this is great 👍🏼